app/services/staff_nomis_checker.rb

```diff
@@ -72,8 +73,27 @@ def slots_unavailable?
     end
   end
 
+  def no_allowance?
```

Comment:
The issue is that this logic is specific to the form in the view, it's not a responsibility at this layer.

The logic of returning false if slots are unavailable, or checking any of the slots errors to decide true or false is specific to how we want the form to work and is not a reusable anywhere else so it makes more sense to be in the decorator.

Maybe the solution is to have methods in the checker like this:

`prisoner_banned?(slot)` and then move the rest of the logic to the decorator.

What do you think?